From 7c3018c7ed4b2e0a089b2bbd4a5150da1590174f Mon Sep 17 00:00:00 2001
From: "apidrift[bot]" <bot@apidrift.dev>
Date: Sun, 30 Aug 2026 23:36:03 +0200
Subject: [PATCH] Migrate deprecated Charges.create to PaymentIntents.create

---
 controllers/api.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/controllers/api.js b/controllers/api.js
index 6a02a61..9963cfa 100644
--- a/controllers/api.js
+++ b/controllers/api.js
@@ -485,12 +485,14 @@ exports.getStripe = (req, res) => {
  */
 exports.postStripe = (req, res) => {
   const { stripeToken, stripeEmail } = req.body;
-  stripe.charges.create(
+  stripe.paymentIntents.create(
     {
       amount: 395,
       currency: 'usd',
-      source: stripeToken,
+      payment_method: stripeToken,
       description: stripeEmail,
+      confirm: true,
+      automatic_payment_methods: { enabled: true, allow_redirects: 'never' }
     },
     (err) => {
       if (err && err.type === 'StripeCardError') {
-- 
2.39.5
